### PR TITLE
CentOS 6.6 libstdc++ fix

### PR DESCRIPTION
**Pull request by twKrash**
Fix for 
https://github.com/flavorjones/chromedriver-helper/issues/25

### CODE REVIEW

**Comment by flavorjones** [comment]
Hi! Apologies for the late response on this PR, I somehow missed this when it came in (I was taking a break from OSS in November).

I have a couple of suggestions I'd like to discuss with respect to this, though.

First, I'd really like to see test coverage for a change like this. This code is complex enough that it's likely to get broken by future changes unless there's test coverage.

Second, I'd prefer to treat environment variables as a first-class aspect of how the commands are run, and use a gem like [with_env](https://github.com/mhs/with_env-rb) to set the environment variables in a block (rather than as part of the command string).

Finally, this particular solution feels brittle to me, though I admit I don't fully understand what your intention is. It looks like there are assumptions here that I feel like we should discuss, I'm going to comment on the commit ...

File: lib/chromedriver/helper.rb [hunk 1]
```diff
@@ -35,6 +35,20 @@ def update
       download true
     end
 
+    def compatibility
+      if File.exists?("/usr/lib64/libstdc++.so.6")
```
Comment (flavorjones) [inline lib/chromedriver/helper.rb]: If this file doesn't exist, does the problem not exist? Can the problem exist without this file being present?

File: lib/chromedriver/helper.rb [hunk 2]
```diff
@@ -35,6 +35,20 @@ def update
...
+        if `strings /usr/lib64/libstdc++.so.6 | grep -e 'GLIBC.*3.4.15'`.empty?
```
Comment (flavorjones) [inline lib/chromedriver/helper.rb]: What's the intention behind specifying `3.4.15`? What should happen if a different glibc version is detected? Is there a more robust way to check for what you want to check for? What if `strings` isn't installed? Why aren't the dots separating the version parts escaped in this regex?

File: lib/chromedriver/helper.rb [hunk 3]
```diff
@@ -35,6 +35,20 @@ def update
...
+          chrome_dir = File.dirname(%x(readlink -f "`which google-chrome`").chomp)
+          path = "#{chrome_dir}:#{chrome_dir}/lib"
```
Comment (flavorjones) [inline lib/chromedriver/helper.rb]: Do we need both of these directories added to LD_LIBRARY_PATH? If so, why?

File: lib/chromedriver/helper.rb [hunk 4]
```diff
@@ -35,6 +35,20 @@ def update
...
+          unless `echo $LD_LIBRARY_PATH`.empty?
```
Comment (flavorjones) [inline lib/chromedriver/helper.rb]: You should not exec in order to check presence/absence of an environment variable. You should use `ENV["LD_LIBRARY_PATH"]` in native Ruby.

File: lib/chromedriver/helper.rb [hunk 5]
```diff
@@ -35,6 +35,20 @@ def update
...
+            path += ":$LD_LIBRARY_PATH"
+          end
+          return "LD_LIBRARY_PATH=#{path} DBUS_SESSION_BUS_ADDRESS=/dev/null"
```
Comment (flavorjones) [inline lib/chromedriver/helper.rb]: The DBUS env var is strange to me, and I'd like to understand why it's here.

Comment (Vanuan) [inline lib/chromedriver/helper.rb]: It's borrowed from https://github.com/SeleniumHQ/docker-selenium/issues/87

Comment (Vanuan) [inline lib/chromedriver/helper.rb]: The reason of this bug is not found, but the solution works.

**Comment by flavorjones** [comment]
Please see the discussion at #25 for an explanation of how I'm addressing this problem.

**Pull request closed without merging**